### ISSUE TITLE
v4.1.x upgrade to v4.2.x haproxy issues

### ISSUE THREAD

**Issue by cloudnull** [rcbops/chef-cookbooks#710]
When performing an upgrade the HAProxy shuts down and refuses to restart due to configuration issues.  This is issue is caused by the configuration file "**vs_quantum-api.cfg**" conflicting with the new configuration file "**vs_neutron-api.cfg**". This happens on the first upgrade `chef-client` run using the v4.2.x cookbooks. To resolve the issue the file "`etc/haproxy/haproxy.d/vs_quantum-api.cfg`" has to be removed prior to performing the upgrade. 

Resolving the issues::

```
rm etc/haproxy/haproxy.d/vs_quantum-api.cfg
service haproxy restart
```

This would be a nice to have in the havana cookbooks but is an easy fix on the users end and once the upgrade is complete is not anything that will need to be done ever again.

**Comment by breu** [comment]
We should really be removing this in the cookbook before we drop the new haproxy conf file.  I don't mind keeping this in 4.2 and then removing it in 4.3

**Comment by klevenstein** [comment]
Documenting as a known issue and also including as a step in the 4.2.1 upgrade procedure. If/when this gets fixed, will just yank the step out of the official procedure.

ETA: documented in the Upgrade Guide.
https://one.rackspace.com/download/attachments/79960660/rackspace-private-cloud-upgrade-20140307.pdf?version=1&modificationDate=1394222265000&api=v2

**Comment by odyssey4me** [comment]
As this is known, why not just delete the '_quantum_' files from the haproxy config directory as part of the recipe?
